conanfile.py

```diff
@@ -80,7 +78,24 @@ def imports(self):
         self.copy("license*", src=".", dst="./licenses",
                   folder=True, ignore_case=True)
 
+    def requirements(self):
+        # Direct dependencies:
+        self.requires("fmt/9.1.0")
+        self.requires("nlohmann_json/3.11.2")
+        self.requires("grpc/1.48.0")
+        self.requires("protobuf/3.21.9")
+        self.requires("paho-mqtt-cpp/1.2.0")
+        self.requires("cpr/1.9.3")
+
+        # Transient dependencies:
+        self.requires("openssl/1.1.1t")
+        # Workaround: Pin recipe revision for transient dependency googleapis for enabling the container build
+        self.requires("googleapis/cci.20221108@#e4bebdfa02f3b6f93bae1d5001b8d439")
+        # Workaround: 
+        self.requires("paho-mqtt-c/1.3.9@#0421671a9f4e8ccfa5fc678cfb160394")
+
     def build_requirements(self):
         # 'build' context (protoc.exe will be available)
-        self.tool_requires("protobuf/3.21.4")
+        self.tool_requires("protobuf/3.21.9")
         self.tool_requires("grpc/1.48.0")
```

Comment:
grpc 1.48.0 is required 3.21.4, therefore we need to update it on 1.50.1

Reply:
This depends on the revision of the Conan recipe. Basically you are right, the latest recipe revision of grpc/1.48.0 refers to protobuf/3.21.4.
But: Conan "isn't the world" and you could decide to use another package manager or plugging library together manually. Means, due to semantic versioning the increase to 3.21.9 is just an increase of the patch (or micro) version of protobuf, so I definitively would expect downward compatible behaviour of it. (Another talk would be if a grpc would require 3.21.9 and we would use 3.21.4. In that situation the grpc lib could expect some certain behaviour available from 3.21.9 onwards, which is maybe not available in 3.21.4. Nevertheless, you'll find differing recipe revisions for grpc/1.50.1 being happy with protobuf 3.21.4 and 3.21.9 ...)

Reply:
grpc 1.50.1 seems to work, will test tomorrow